tests/test_swapped_models.py

```diff
@@ -0,0 +1,165 @@
+import importlib
+import os
+import subprocess
+
+from django.conf import settings
+from django.db import migrations
+from django.test import TestCase
+from django.test.utils import override_settings
+
+
+class TestSwappableModels(TestCase):
+    """
+    This test verifies that it is possible to install `django-scheduler`
+    and then swap its models.
+    """
+    settings_path = os.path.join(settings.BASE_DIR, 'tests', 'test_project', 'settings.py')
+    updated_settings_path = os.path.join(settings.BASE_DIR, 'tests', 'test_project', 'updated_settings.py.tpl')
+
+    def setUp(self):
+        super(TestSwappableModels, self).setUp()
+        # cleanup any possible leftovers before running
+        self._cleanup()
+        # for backup purposes
+        with open(self.settings_path) as settings_file:
+            self.settings_content = settings_file.read()
+
+    def tearDown(self):
+        # cleanup leftovers after running
+        self._cleanup()
+        # reset the settings to their previous value
+        with open(self.settings_path, 'w') as settings_file:
+            settings_file.write(self.settings_content)
+        super(TestSwappableModels, self).tearDown()
+
+    def _cleanup(self):
+        """
+        Remove possibly generated migration files.
+        """
+        migration_file = os.path.join(
+            settings.BASE_DIR,
+            'tests',
+            'test_project',
+            'migrations',
+            '0001_initial.py'
+        )
+        try:
+            os.remove(migration_file)
+        except OSError:
+            # it's ok
+            pass
+
+    def _check_command_return(self, command_name, popen):
+        """
+        Check bash command return code, and fail if the return
+        code is non-zero. It takes a `command_name` argument to
+        make the output human-readable, and a `Popen` instance
+        in order to inspect it.
+        """
+        if popen.returncode != 0:
+            self.fail(
+                'Command {} failed, return code {}\n'
+                'stdout: {}\n'
+                'sterr: {}\n'.format(
+                    command_name,
+                    popen.returncode,
+                    popen.stdout,
+                    popen.stderr
+                )
+            )
+
+    def _execute_makemigrations(self):
+        """
+        Execute `python -m django makemigrations test_project --settings=tests.test_project.settings`.
+        """
+        popen = subprocess.Popen(
+            [
+                'python',
```

Comment:
This should probably use [`sys.executable`](https://docs.python.org/3/library/sys.html#sys.executable)

Reply:
9661a38 ✅